python/hsfs/storage_connector.py

```diff
@@ -587,6 +588,51 @@ def prepare_spark(self, path: Optional[str] = None):
     def _get_path(self, sub_path: str):
         return os.path.join(self.path, sub_path)
 
+    def read(
+        self,
+        query: str = None,
+        data_format: str = None,
+        options: dict = {},
+        path: str = "",
+    ):
```

Comment:
This method is wrong. It should behave like the read method from the S3 connector https://github.com/logicalclocks/feature-store-api/blob/30443492fcabdc21a8c0f9d458c5dd7f9741bc5c/python/hsfs/storage_connector.py#L266

The `path` parameter should be the path within the ADLS container and not the full path including the container name, account id and the domain.
Otherwise there is absolutely no point of creating the storage connector if users need to provide again this information while calling the `read()` method.

Additionally the difference between Gen1 and Gen2 are not well known, so you are asking the user to know this. Again what's the point of the storage connector then?

Please take a look at the S3 implementation and do the same.

Reply:
Alright, looking at the docs and implementation I thought path should follow gen1 and gen2 as specified on the container option. I made changes to read from the default container specified if the path is not a full URI and removed validation for gen1 and gen2 path format.